content/spin/distributing-apps.md

```diff
@@ -1,104 +1,133 @@
 
-To distribute applications, we first need a Bindle registry. You can [install Bindle v0.8 release](https://github.com/deislabs/bindle/tree/main/docs#from-the-binary-releases), or use the [`autobindle`](https://marketplace.visualstudio.com/items?itemName=fermyon.autobindle) VS Code extension (through the `Bindle: Start` command):
+## Running Published Applications
+
+To run a published application from a registry, use `spin up -f` and pass the registry reference:
 
 <!-- @nocpy -->
 
 ```bash
-$ bindle-server --address 127.0.0.1:8000 --directory . --unauthenticated
+$ spin up -f ghcr.io/alyssa-p-hacker/hello-world:v1
 ```
 
-Let's push the application from the [quickstart](./quickstart.md) to the registry:
+> Remember that if the artifact is private you will need to be logged in, with permission to access it.
 
-<!-- @selectiveCpy -->
+### Running Published Applications by Digest
+
+Registry versions are mutable; that is, the owner of an application can change which build the `:v1` label points to at any time.  If you want to run a specific build of the package, you can refer to it by _digest_.  This is similar to a Git commit hash: it is immutable, meaning the same digest always gets the exact same data, no matter what the package owner does.  To do this, use the `@sha256:...` syntax instead of the `:v...` syntax:
 
 ```bash
-$ export BINDLE_URL=http://localhost:8000/v1
-$ spin bindle push --file spin.toml
-pushed: spin-hello-world/1.0.0
+$ spin up -f ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19
 ```
 
-Now we can run the application using `spin up` directly from the registry:
+### Pulling a Published Application
 
-<!-- @selectiveCpy -->
+`spin up` automatically downloads the application from the registry. If you want to manually download the application, without running it, use the `spin registry pull` command:
+
+<!-- @nocpy -->
 
 ```bash
-$ spin up --bindle spin-hello-world/1.0.0
+$ spin registry pull ghcr.io/alyssa-p-hacker/hello-world:v1
+$ spin registry pull ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19
 ```
 
-> A known error when running an application (with a high number of static assets) from Bindle on macOS is around [too many open files](https://github.com/fermyon/spin/issues/180). This issue can be worked around by setting a higher `ulimit -n` for the current shell
-> session.
+> Downloaded applications are cached. When run, or pulled again, Spin checks to see if they have changed from the cached copy, and downloads only the changes if any.
+
+## Signing Spin Applications and Verifying Signatures
+
+Because Spin uses the container registry standards to distribute applications, it can also take advantage of tooling built around those standards.  Here is an example of using [Cosign and Sigstore](https://docs.sigstore.dev/cosign/overview/) to sign and verify a Spin application:
+
+<!-- @nocpy -->
+
+```bash
+# Push your Spin application to any registry that supports the OCI registry artifacts,
```

Comment:
nano-nit: Elsewhere we are in the habit of supplying informational/instructional test in markdown, interspersed with bash snippets.  We might consider updating this section to follow suit.

Reply:
I am going to punt on this for now if that is okay.